src/sentry/web/frontend/base.py

```diff
@@ -189,15 +189,33 @@ def get_active_project(self, request: Request, organization, project_slug):
     def redirect_to_org(self, request: Request):
         from sentry import features
 
+        using_customer_domain = request and is_using_customer_domain(request)
+
         # TODO(dcramer): deal with case when the user cannot create orgs
         if self.active_organization:
-            url = Organization.get_url(self.active_organization.organization.slug)
+            current_org_slug = self.active_organization.organization.slug
+            url = Organization.get_url(current_org_slug)
         elif not features.has("organizations:create"):
             return self.respond("sentry/no-organization-access.html", status=403)
         else:
+            org_exists = False
             url = "/organizations/new/"
-            if request.subdomain:
+            if using_customer_domain:
                 url = absolute_uri(url)
+
+            if using_customer_domain and request.user and request.user.is_authenticated:
+                organizations = organization_service.get_organizations(
+                    user_id=request.user.id, scope=None, only_visible=True
+                )
```

Comment:
Why do we need to fetch all the organizations? Are you trying to keep users out that have no organizations, or just anonymous users?

Reply:
I had to patch this edge case because when a user accesses `orgslug.sentry.io`, then `self.active_organization` would either be `orgslug` (if they're an org member), or nothing at all. 

In case the user is not an org member, we check if the user has any organizations they're a member of, and if so, we redirect them to the login page. Otherwise, we redirect them to creating a new organization.

I agree that we shouldn't need to fetch all the organizations. I had plan to simplify this later on with an improved API for `organization_service`.